### ISSUE TITLE
Upgrading to sp-client-custom-fields new version

### ISSUE THREAD

**Issue by natelabo** [OlivierCC/spfx-40-fantastics#14]
I'm not sure if this needs to be done or not. But I was updating to the latest sp-client-custom-fields.

I've added "key" to the fields array. But I don't believe it's being referenced by the function. My picker data comes up empty.

Do I also need to update the ...WebPartProps.ts file?

**Comment by OlivierCC** [comment]
Which kind of picker do you use from the sp-client-custom-fields? It's mandatory to upgrade this lib, but in fact it's better with the latest version (1.3.3) because of update in Office UI Fabric in SharePoint Online. The only problem is that for the moment I didn't test the Fantastic 40 Web Parts with this version. I will do that as soon as possible. Be careful to use the 1.3.3 version of sp-client-custom-fields and not other (1.3.X), this version is very more stable. I will post here more info about this update on my test env soon. Thanks again @natelabo for your reports

**Comment by natelabo** [comment]
I also attempted to update the Office UI Fabric...

Here's an example of the code I updated...

NewsCarousel:
`                PropertyFieldCustomList('items', {
                  label: strings.Items,
                  value: this.properties.items,
                  headerText: strings.ManageItems,
                  fields: [
                    {id: 'NewsCarouselTitle', title: 'Title', required: true, type: CustomListFieldType.string },
                    {id: 'NewsCarouselEnable', title: 'Enable', required: true, type: CustomListFieldType.boolean },
                    /*{ title: 'Start Date', required: false, type: CustomListFieldType.date },
                    { title: 'End Date', required: false, type: CustomListFieldType.date },
                    */
                    {id: 'NewsCarouselDescription', title: 'Description', required: false, hidden: true, type: CustomListFieldType.string },
                    {id: 'NewsCarouselPicture', title: 'Picture', required: true, hidden: true, type: CustomListFieldType.string },
                    {id: 'NewsCarouselURL', title: 'Link Url', required: true, hidden: true, type: CustomListFieldType.string }
                  ],
`

When the picker is pulled up it looks like this the attached picture.

![picker](https://cloud.githubusercontent.com/assets/19813240/25037261/1b3055b2-20c6-11e7-8db9-4454d34b346a.PNG)

Let me know how I can help. This package is amazing and I'm happy to help in any way I can... Thanks!

**Comment by OlivierCC** [comment]
Yes, the new 'id' is a big change with new version in the PropertyFieldCustomList custom field.

My advices to work fine with it:
- Big bugs in the 1.3.2 in this control, be careful to use the 1.3.3
- Put the same value in 'id' that in 'title' => previously the title was use to create returned object properties

You should try something like that:

PropertyFieldCustomList('items', { label: strings.Items, value: this.properties.items, headerText: strings.ManageItems, fields: [ {id: 'Title', title: 'Title', required: true, type: CustomListFieldType.string }, {id: 'Enable', title: 'Enable', required: true, type: CustomListFieldType.boolean }, {id: 'Description', title: 'Description', required: false, hidden: true, type: CustomListFieldType.string }, {id: 'Picture', title: 'Picture', required: true, hidden: true, type: CustomListFieldType.string }, {id: 'Link Url', title: 'Link Url', required: true, hidden: true, type: CustomListFieldType.string } ],

**Comment by OlivierCC** [comment]
Hi @natelabo, the new version 1.0.3.0 of these webparts includes the 1.3.3 version of the sp-client-custom-fields and is working fine. Let me know if you have still problem with this lib or if I can close this bug.

Thanks, regards,

Olivier
